src/tasks/pop.ts

```diff
@@ -1,11 +1,40 @@
+import assign from "../util/assign";
+import prefixKeys from "../util/prefixKeys";
+
 import Task from "./task";
+import { asyncGetEntry, normalizeEntry } from "../lib/resourceTiming";
 
 export default class Pop extends Task {
   public constructor(config: Config, taskData: TaskData) {
     super(config, taskData);
   }
 
-  protected test(): Promise<any> {
-    return Promise.resolve();
+  private fetchObjectAndId(): Promise<string> {
+    return fetch(this.data.resource).then(
+      (res): string => res.headers.get("X-Datacenter") || ""
+    );
+  }
+
+  protected test(): Promise<ResourceTimingEntry> {
+    let subjectId: string;
+
+    return Promise.all([
+      this.fetchObjectAndId(),
+      asyncGetEntry(this.data.resource)
+    ])
+      .then(
+        ([id, entry]): ResourceTimingEntry => {
+          subjectId = id;
+          return normalizeEntry(entry);
+        }
+      )
+      .then(
+        (timing: ResourceTimingEntry): ResourceTimingEntry => {
+          // eslint-disable-next-line @typescript-eslint/camelcase
+          const meta = { id: subjectId, attempted_id: this.data.id };
+          const data = prefixKeys(assign(meta, timing), "subject_");
+          return data;
+        }
+      );
```

Comment:
This code seems more complex than it needs to be with the closure and the two `.then()` statements. Can't you just do:
```js
    return Promise.all([
      this.fetchObjectAndId(),
      asyncGetEntry(this.data.resource)
    ])
      .then(
        ([id, entry]): ResourceTimingEntry => {
          // eslint-disable-next-line @typescript-eslint/camelcase
          const meta = { id, attempted_id: this.data.id };
          const timing = normalizeEntry(entry);
          return prefixKeys(assign(meta, timing), "subject_");
        }
      );
```
?